in attributes filter allow selecting from which connectors attributes are shown

ztNFny opened the pull request:
Allow filtering attributes by the connector that uses them in filters "Cache Attributes" dialog.
GC uses 70 attributes, OC (okapi) 89 for a total of 124. For a GC-only cacher that means there are currently 54 unused icons shown in the list.

This feature allows the user to have only those attributes relevant for him visible, controllable with buttons in the filter dialog itself.
The default setting is based on enabled connectors (I'm open to removing this depending on feedback - this is where it started but with a GUI element I'm not too fixed on it)

This also fixes #12425 and #12433 along with some layout changes to "Attributes Overview" and attributes text view in cache details

GC only:
![image](https://user-images.githubusercontent.com/1258173/147686549-d2b1354f-b453-417d-9490-b54fe806e06c.png)

OC only:
![image](https://user-images.githubusercontent.com/1258173/147686564-71c21117-b61c-4b69-a235-4d8f6c29793d.png)

GC + OC:
![image](https://user-images.githubusercontent.com/1258173/147686583-a105d10e-d9db-4f1e-8984-1cef0e7af159.png)

Attributes text view in cache:
![image](https://user-images.githubusercontent.com/1258173/147692269-c6bdf6a8-7441-44fe-bdce-dee61175b4d0.png)

Attribute overview:
![image](https://user-images.githubusercontent.com/1258173/147692301-4a28cca7-7a79-4e04-897f-e5c5c2cbf2b2.png)

ztNFny commented:
For those trying to review the changes please take a look at https://github.com/ztNFny/cgeo/commits/issue/12311_filterAttributes , starting with commit "filter list of attributes by connector in filter"

I had to squash the changes into a new branch as the merge was a nightmare

ztNFny commented:
retest this please

ztNFny commented:
Test failure as https://www.opencaching.us is currently down 🤨

ztNFny commented:
retest this please

ztNFny commented:
rebased "Attributes Overview" to work with bottomSheetDialogWithActionbar
![2022-01-03_09-39-57](https://user-images.githubusercontent.com/1258173/147912162-00e8c75d-47e3-4071-a2a0-15e20b2dc366.png)

ztNFny commented:
retest this please

moving-bits commented:
There seems to be an error in the source code:

```
/main/src/cgeo/geocaching/ui/CacheListAdapter.java:312: error: cannot find symbol
((WrappingGridView) v.findViewById(R.id.attributes_grid)).setAdapter(new AttributesGridAdapter((Activity) context, a, null));
symbol:   variable a
location: class CacheListAdapter
```

ztNFny commented:
> There seems to be an error in the source code:

Strange, that's a merge remnant I already fixed this morning, and viewing the commit online confirms the change is in there:
![image](https://user-images.githubusercontent.com/1258173/147965410-bfd68d1a-7642-4c79-a67c-7662dc5acbd4.png)

I've reverted and re-pushed the commit, let's see.
